IPython/core/completer.py

```diff
@@ -495,10 +495,14 @@ def rectify_completions(text: str, completions: _IC, *, _debug=False)->_IC:
         print('IPython.python matches have extras:', diff)
 
 
+# Keep two lists for delimiters, first one works for filenames containing /, -,
+# and ~ (for expansion)
 if sys.platform == 'win32':
-    DELIMS = ' \t\n`!@#$^&*()=+[{]}|;\'",<>?'
+    DELIMS = [' \t\n`!@#$^&*()=+[{]}|;\'",<>?',
+              ' \t\n`!@#$^&*()=+[{]}|;\'",<>?-~/']
 else:
-    DELIMS = ' \t\n`!@#$^&*()=+[{]}\\|;:\'",<>?'
+    DELIMS = [' \t\n`!@#$^&*()=+[{]}\\|;:\'",<>?',
+              ' \t\n`!@#$^&*()=+[{]}\\|;:\'",<>?-~/']
```

Comment:
Technically API change. DELIMS is exported and is a string, we can't just change the type to a list just after a major release, we'll have to figure out a way to get Backward  compat.